File: algorithms/sort/comparison/bubble/java/BubbleSort.java
public class BubbleSort {
    public void sort(int[] a) {
        for(int i = 1; i < a.length; i++) {
            for(int j = 0; j < a.length - i; j++) {
                if(a[j] > a[j+1]) {
                    var tmp = a[j];
                    a[j] = a[j+1];
                    a[j+1] = tmp;
                }
            }
        }
    }
}

File: algorithms/sort/comparison/bubble/kotlin/BubbleSort.kt
class BubbleSort {
    fun sort(a: IntArray) {
        for(i in 1 until a.size) {
            for(j in 0 until a.size - i) {
                if(a[j] > a[j+1]) {
                    val tmp = a[j]
                    a[j] = a[j+1]
                    a[j+1] = tmp;
                }
            }
        }
    }
}
